src/client_state.cpp

```diff
@@ -262,11 +262,15 @@ int wsrep::client_state::after_statement()
     debug_log_state("after_statement: enter");
     assert(state() == s_exec);
     assert(mode() == m_local);
-<<<<<<< HEAD
 
-    if ((transaction_.active() &&
-         transaction_.state() == wsrep::transaction::s_must_abort) ||
-        (transaction_.force_bf_rollback()))
+    // Handle BF Abort of the local ROLLBACK TO SAVEPOINT
+    // (PXC-2930) (commit 2d1e7abc)
+    //
+    // If local transaction executing rollback to savepoint is interrupted by a
+    // local running high priority transaction (DDL) then semantics should
+    // cause local transaction to completely abort (and not limit to rollback
+    // of savepoint only). This is set in ha_rollback_to_savepoint().
+    if (transaction_.force_bf_rollback())
```

Comment:
No. This change was done in favor of the upstream fix made in commit 62bd4080
```
commit 62bd40800f1c39a82c5663b7b3e3afbab39332aa
Author: Daniele Sciascia <daniele.sciascia@galeracluster.com>
Date:   Mon Jul 17 14:49:33 2023 +0200

    Remove duplicate logic from after_statement
    
    Remove bf abort handling `client_state::after_statement()`, since the
    same logic already appears later in `transaction::after_statement()`.
    Also, introduce `transaction::after_statement()` overload which takes
    a lock.
```

which removes the duplication of code in the after_statement() hook.

Earlier we had same code in two places.

The first one was in `wsrep::client_state::after_statement()`
```
  264     if ((transaction_.active() &&                                                                                                                                                                            
  265     |   |transaction_.state() == wsrep::transaction::s_must_abort) ||                                                                                                                                        
  266     |   (transaction_.force_bf_rollback()))                                                                                                                                                                  
  267     {                                                                                                                                                                                                        
  268     |   lock.unlock();                                                                                                                                                                                       
  269     |   client_service_.bf_rollback();                                                                                                                                                                       
  270     |   lock.lock();                                                                                                                                                                                         
  271     |   assert(transaction_.state() == wsrep::transaction::s_aborted);                                                                                                                                       
  272     |   // Error may be set already. For example, if fragment size                                                                                                                                           
  273     |   // exceeded the maximum size in certify_fragment(), then                                                                                                                                             
  274     |   // we already have wsrep::e_error_during_commit                                                                                                                                                      
  275     |   if (current_error() == wsrep::e_success)                                                                                                                                                             
  276     |   {                                                                                                                                                                                                    
  277     |   |   override_error(wsrep::e_deadlock_error);                                                                                                                                                         
  278     |   }                                                                                                                                                                                                    
  279     }                                                                                                                                                                                                        
  280     lock.unlock();                                                                                                                                                                                           
  281                                                                                                                                                                                                              
  282     (void)transaction_.after_statement();
```

and the second one in `wsrep::transaction::after_statement()`
```
   858     case s_must_abort:                                                                                                                                                                                      
   859     case s_cert_failed:                                                                                                                                                                                     
   860     |   // Error may be set already. For example, if fragment size                                                                                                                                          
   861     |   // exceeded the maximum size in certify_fragment(), then                                                                                                                                            
   862     |   // we already have wsrep::e_error_during_commit                                                                                                                                                     
   863     |   if (client_state_.current_error() == wsrep::e_success)                                                                                                                                              
   864     |   {                                                                                                                                                                                                   
   865     |   |   client_state_.override_error(wsrep::e_deadlock_error);                                                                                                                                          
   866     |   }                                                                                                                                                                                                   
   867     |   lock.unlock();                                                                                                                                                                                      
   868     |   ret = client_service_.bf_rollback();                                                                                                                                                                
   869     |   lock.lock();                                                                                                                                                                                        
   870     |   if (state() != s_must_replay)                                                                                                                                                                       
   871     |   {                                                                                                                                                                                                   
   872     |   |   break;                                                                                                                                                                                          
   873     |   }                                                                                                                                                                                                   
   874     |   // Continue to replay if rollback() changed the state to s_must_replay                                                                                                                              
   875     |   WSREP_FALLTHROUGH;                  
```